tests/1_host_configuration.sh

```diff
@@ -59,18 +59,19 @@ check_1_1_2() {
 
 check_1_2() {
   logit ""
-  id_1_2="1.2"
-  desc_1_2="Linux Hosts Specific Configuration"
-  check_1_2="$id_1_2 - $desc_1_2"
-  info "$check_1_2"
+  local id="1.2"
+  local desc="Linux Hosts Specific Configuration"
+  local check="$id - $desc"
+  info "$check"
 }
 
 # 1.2.1
 check_1_2_1() {
-  id_1_2_1="1.2.1"
-  desc_1_2_1="Ensure a separate partition for containers has been created (Scored)"
-  check_1_2_1="$id_1_2_1 - $desc_1_2_1"
-  starttestjson "$id_1_2_1" "$desc_1_2_1"
+  local id="1.2.1"
+  local desc="Ensure a separate partition for containers has been created (Scored)"
+  local remediation="For new installations, you should create a separate partition for the /var/lib/docker mount point. For systems that have already been installed, you should use the Logical Volume Manager (LVM) within Linux to create a new partition."
```

Comment:
I really like the `remediation` stuff, but it gets very very spammy. Collect the all and present it as a list in the end?


```sh
[INFO] 1.2 - Linux Hosts Specific Configuration
[WARN] 1.2.1 - Ensure a separate partition for containers has been created (Scored)
[INFO]        * Remediation: For new installations, you should create a separate partition for the /var/lib/docker mount point. For systems that have already been installed, you should use the Logical Vol
ume Manager (LVM) within Linux to create a new partition.
[INFO] 1.2.2  - Ensure only trusted users are allowed to control Docker daemon (Scored)
[INFO]        * docker:x:997:vagrant   
[WARN] 1.2.3  - Ensure auditing is configured for the Docker daemon (Scored)
[INFO]        * Remediation: Install and configure auditd using the following command: sudo apt-get install auditd. Add the following line to the /etc/audit/rules.d/audit.rules file: -w /usr/bin/dockerd -
k docker. Then restart the audit daemon using the following command: service auditd restart.
[INFO]        * Remediation impact: Because the audit can generate large log files, you need to make sure that they are rotated and archived periodically. It is a good idea to create a separate partition 
for audit logs to avoid filling up other critical partitions.
[WARN] 1.2.4  - Ensure auditing is configured for Docker files and directories - /var/lib/docker (Scored)
[INFO]        * Remediation: Install and configure auditd using the following command: sudo apt-get install auditd. Add the following line to the /etc/audit/rules.d/audit.rules file: -w /var/lib/docker -k
 docker. Then restart the audit daemon using the following command: service auditd restart.
[INFO]        * Remediation impact: Because the audit can generate large log files, you need to make sure that they are rotated and archived periodically. It is a good idea to create a separate partition 
for audit logs to avoid filling up other critical partitions.
[WARN] 1.2.5  - Ensure auditing is configured for Docker files and directories - /etc/docker (Scored)
[INFO]        * Remediation: Install and configure auditd using the following command: sudo apt-get install auditd. Add the following line to the /etc/audit/rules.d/audit.rules file: -w /etc/docker -k doc
ker. Then restart the audit daemon using the following command: service auditd restart.
[INFO]        * Remediation impact: Because the audit can generate large log files, you need to make sure that they are rotated and archived periodically. It is a good idea to create a separate partition 
for audit logs to avoid filling up other critical partitions.
[WARN] 1.2.6  - Ensure auditing is configured for Docker files and directories - docker.service (Scored)
[INFO]        * Remediation: Install and configure auditd using the following command: sudo apt-get install auditd. Add the following line to the /etc/audit/rules.d/audit.rules file: -w /lib/systemd/syste
m/docker.service -k docker. Then restart the audit daemon using the following command: service auditd restart.
[INFO]        * Remediation impact: Because the audit can generate large log files, you need to make sure that they are rotated and archived periodically. It is a good idea to create a separate partition 
for audit logs to avoid filling up other critical partitions.
[WARN] 1.2.7  - Ensure auditing is configured for Docker files and directories - docker.socket (Scored)
[INFO]        * Remediation: Install and configure auditd using the following command: sudo apt-get install auditd. Add the following line to the /etc/audit/rules.d/audit.rules file: -w /lib/systemd/syste
m/docker.socket -k docker. Then restart the audit daemon using the following command: service auditd restart.
[INFO]        * Remediation impact: Because the audit can generate large log files, you need to make sure that they are rotated and archived periodically. It is a good idea to create a separate partition 
for audit logs to avoid filling up other critical partitions.
[WARN] 1.2.8  - Ensure auditing is configured for Docker files and directories - /etc/default/docker (Scored)
[INFO]        * Remediation: Install and configure auditd using the following command: sudo apt-get install auditd. Add the following line to the /etc/audit/rules.d/audit.rules file: -w /etc/default/docke
r -k docker. Then restart the audit daemon using the following command: service auditd restart.
[INFO]        * Remediation impact: Because the audit can generate large log files, you need to make sure that they are rotated and archived periodically. It is a good idea to create a separate partition 
for audit logs to avoid filling up other critical partitions.
[INFO] 1.2.9  - Ensure auditing is configured for Docker files and directories - /etc/sysconfig/docker (Scored)
[INFO]        * File not found
```

Reply:
I fully agree with you. I think it's a good idea to collect them and print them all at the end.